### PR TITLE
Drop the support for Python 3.7 and update package metadata for Python 3.13 support

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation
Closes https://github.com/optuna/optuna/issues/4586

## Description of the changes
This PR updates followings to drop the support for Python 3.7.

* CI settings
* README and documentation
* Package metadata

For any remaining TODO comments in the implementation and tests, I will open a follow-up PR after this one is merged: https://github.com/optuna/optuna/compare/master...c-bata:optuna:resolve-todo-comments-for-python3.7?expand=1

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,12 +15,12 @@ classifiers = [
   "Intended Audience :: Developers",
   "License :: OSI Approved :: MIT License",
   "Programming Language :: Python :: 3",
-  "Programming Language :: Python :: 3.7",
   "Programming Language :: Python :: 3.8",
   "Programming Language :: Python :: 3.9",
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
```
Comment (c-bata) [inline pyproject.toml:23]: This is a follow-up changes for https://github.com/optuna/optuna/pull/5691

**Comment by c-bata** [comment]
@nabenabe0928 @gen740 Could you review this PR? Regarding the branch protection rules on the master branch, let me update them after you approve.

File: .github/workflows/tests-storage.yml [hunk 2]
```diff
@@ -23,16 +23,18 @@ jobs:
...
-          - test-trigger-type: ""
-            python-version: "3.8"
           - test-trigger-type: ""
             python-version: "3.9"
           - test-trigger-type: ""
             python-version: "3.10"
+          - test-trigger-type: ""
+            python-version: "3.11"
+          - test-trigger-type: ""
+            python-version: "3.12"
```
Comment (nabenabe0928) [inline .github/workflows/tests-storage.yml:37]: [Question] Don't we test Python 3.12 here?

**Review by nabenabe0928 — COMMENTED** [review]
Mostly LGTM.

Here are the files that need to be modified in the followups of this PR:
- [ ] CONTRIBUTING.md
- [ ] optuna/importance/_ped_anova/evaluator.py
- [ ] optuna/samplers/_nsgaiii/_elite_population_selection_strategy.py
- [ ] tests/artifacts_tests/test_boto3.py
- [ ] tests/samplers_tests/tpe_tests/test_truncnorm.py

File: .github/workflows/tests-storage.yml (hunk 2 above)
Comment (c-bata) [reply .github/workflows/tests-storage.yml:37]: Unlike `tests (3.13)`, which skips certain test scenarios that depend on PyTorch, `tests-storage (3.13)` can cover all storage scenarios. So I updated this file to test with Python 3.8 (oldest) and 3.13 (latest).

**Review by nabenabe0928 — APPROVED** [review]
LGTM:)

**Review by gen740 — CHANGES_REQUESTED** [review]
Almost LGTM!
I noticed that the following TODO comment can be removed.
https://github.com/optuna/optuna/blob/c014b9d01131cfffd3a7b1ef6b323373668c9ac9/optuna/importance/_ped_anova/evaluator.py#L53-L55
and the following pytest.mark can also be removed
https://github.com/optuna/optuna/blob/c014b9d01131cfffd3a7b1ef6b323373668c9ac9/tests/samplers_tests/tpe_tests/test_truncnorm.py#L16-L18.

**Comment by c-bata** [comment]
@gen740 Can I work on them in the following pull request?

> For any remaining TODO comments in the implementation and tests, I will open a follow-up PR after this one is merged: https://github.com/optuna/optuna/compare/master...c-bata:optuna:resolve-todo-comments-for-python3.7?expand=1

**Review by gen740 — APPROVED** [review]
> For any remaining TODO comments in the implementation and tests, I will open a follow-up PR after this one is merged

Oh, I missed this comment.
LGTM!

**Pull request merged**